### PR TITLE
fix(ui5-combobox): announce selected item

### PR DESCRIPTION
**Pull request by ivoplashkov**
FIXES: #2654

### CODE REVIEW

File: packages/main/src/ComboBox.js [hunk 1]
```diff
@@ -722,6 +728,12 @@ class ComboBox extends UI5Element {
 		this._itemFocused = true;
 	}
 
+	announceSelectedItem(indexOfItem) {
+		const itemPositionText = this.i18nBundle.getText(LIST_ITEM_POSITION, [indexOfItem + 1], [this._filteredItems.length]);
```
Comment (elenastoyanovaa) [inline packages/main/src/ComboBox.js:732]: you should add also "Selected"

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]
handleArrowKeyPress will not cover all cases, for example: IF you press Alt+Down and open the combobox, the focus should go to the list (currently not working, but we will fix it in future with the KBH) and the announcement should be present also. Now you are covering only the scenario, when the user navigates inside the list.

File: packages/main/test/specs/ComboBox.spec.js [hunk 2]
```diff
@@ -328,3 +328,20 @@ describe("General interaction", () => {
...
+	it ("Announce item on selection", () => {
+		const combo = $("#combo");
+		const arrow = combo.shadow$("[input-icon]");
+		const input = combo.shadow$("#ui5-combobox-input");
+		const invisibleMessageSpan = $(".ui5-invisiblemessage-polite");
+		const itemAnnouncement = "List item 1 of 11";
+
+		arrow.click();
+
+		input.keys("ArrowDown");
```
Comment (elenastoyanovaa) [inline packages/main/test/specs/ComboBox.spec.js:343]: check also for another arrow down, for arrow up and also that the span is empty when the visual focus is not on the list item.

**Review by elenastoyanovaa — COMMENTED** [review]
Have you checked the reason for the double announcement? Since it is not present in UI5 I assume that this is a bug in the Web Components. If it can't be fixed in this PR, please open an issue.

File: packages/main/src/ComboBox.js (hunk 1 above)
Comment (ivoplashkov) [reply packages/main/src/ComboBox.js:733]: Done.

File: packages/main/test/specs/ComboBox.spec.js (hunk 2 above)
Comment (ivoplashkov) [reply packages/main/test/specs/ComboBox.spec.js:350]: Done.

**Comment by ivoplashkov** [comment]
> handleArrowKeyPress will not cover all cases, for example: IF you press Alt+Down and open the combobox, the focus should go to the list (currently not working, but we will fix it in future with the KBH) and the announcement should be present also. Now you are covering only the scenario, when the user navigates inside the list.

This case would be covered after the KBH implementation. Thе announcement logic should be moved to the _keydown handler, moreover some logic for the currentItem and currentIndex should be moved to getters.

> Have you checked the reason for the double announcement? Since it is not present in UI5 I assume that this is a bug in the Web Components. If it can't be fixed in this PR, please open an issue.

The first announcement comes from the change of the value in the input field and the second announcement comes from selecting the text. This behaviour is not present in UI5, which makes me think that maybe there is some kind of focus traversing that I am not aware of. I will open a seperate issue for this.

**Review by elenastoyanovaa — COMMENTED** [review]
Didn't we talked to clear the invisible message on focus out?

**Review by elenastoyanovaa — APPROVED** [review]

**Pull request merged**